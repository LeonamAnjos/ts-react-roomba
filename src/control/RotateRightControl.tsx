import React, { useContext } from "react";
import { RoombaContext } from "../contexts/RoombaContext";

const RotateRightControl = () => {
  const { rotateRight } = useContext(RoombaContext);

  return <button onClick={rotateRight}>Rotate to the right</button>;
};

export default RotateRightControl;
