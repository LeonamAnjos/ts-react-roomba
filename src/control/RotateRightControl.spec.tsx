import { render, screen, fireEvent } from "@testing-library/react";
import { ReactNode, useContext } from "react";
import RoombaContextProvider, {
  RoombaContext,
} from "../contexts/RoombaContext";
import RotateRightControl from "./RotateRightControl";

type TestRotateRightControlProps = {
  spyRotateRight: () => void;
};

const TestRotateRightControl = ({
  spyRotateRight,
}: TestRotateRightControlProps) => {
  const state = useContext(RoombaContext);
  state.rotateRight = spyRotateRight;

  return <RotateRightControl />;
};

describe("<RotateRightControl />", () => {
  const customRender = (children: ReactNode) => {
    return render(
      <RoombaContextProvider size={3}>{children}</RoombaContextProvider>
    );
  };
  it("SHOULD render", () => {
    const { container } = customRender(<RotateRightControl />);

    expect(container).toMatchSnapshot();
  });

  it("SHOULD call moveForward WHEN button is clicked", () => {
    const spy = jest.fn();
    customRender(<TestRotateRightControl spyRotateRight={spy} />);

    const element = screen.getByTestId("rotate-right-button");

    fireEvent.click(element);
    expect(spy).toHaveBeenCalledTimes(1);

    fireEvent.click(element);
    expect(spy).toHaveBeenCalledTimes(2);
  });
});
