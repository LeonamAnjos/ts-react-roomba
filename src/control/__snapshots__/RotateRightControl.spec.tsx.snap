// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RotateRightControl /> SHOULD render 1`] = `
<div>
  <button
    data-testid="rotate-right-button"
  >
    Rotate to the right
  </button>
</div>
`;
